src/lib/Components/Home/ArtistRails/ArtistRail.tsx

```diff
@@ -28,29 +30,24 @@ interface State {
   loadFailed: boolean
 }
 
-class ArtistRail extends React.Component<Props, State> {
-  constructor(props) {
-    super(props)
-    this.state = {
-      artists: [],
-      loadFailed: false,
-    }
-  }
+export class ArtistRail extends Component<Props, State> {
+  inflightRequest: Disposable
 
-  componentDidMount() {
-    if (this.props.relay) {
-      this.props.relay.refetch({ __id: this.props.rail.__id, fetchContent: true }, null, error => {
-        if (error) {
-          console.error("ArtistRail.tsx", error.message)
+  state = {
+    artists: [],
+    loadFailed: false,
+  }
 
-          this.setState({
-            loadFailed: true,
-          })
-        }
-      })
+  componentWillMount() {
+    if (this.props.registerRailModule) {
+      this.props.registerRailModule(this)
     }
   }
 
+  async componentDidMount() {
+    await this.refreshData()
+  }
```

Comment:
I don’t understand what this is effectively achieving, it’s not blocking the thread in any way, right?

Reply:
Nope not at all, it's just kicking off the fetch

Reply:
Ok. My brain seems to have a little problem parsing this code without the presence of any other code around it, probably just because of the word ‘await’, as if it’s waiting for something.